### PR TITLE
Add support for changing the ownership of the vault secret file

### PR DESCRIPTION
**Pull request by shuai-hao-hs**

### CODE REVIEW

File: secrets/json_secrets.go [hunk 1]
```diff
@@ -111,6 +118,13 @@ func writeField(secret config.SecretType, kvSecrets []briefcase.SimpleSecret, fi
 		if err != nil {
 			return fmt.Errorf("failed writing secret to file %q: %w", field.Output, err)
 		}
+
+		if secret.Owner != nil {
+            err = os.Chown(field.Output, secret.Owner , secret.Owner)
```
Comment (tom-hadlaw-hs) [inline secrets/json_secrets.go:123]: formatting seems off here, use `go fmt`

File: secrets/json_secrets.go [hunk 2]
```diff
@@ -52,6 +52,13 @@ func WriteComposite(composite config.CompositeSecretFile, cache briefcase.Secret
 		}
 	}
 
+	if composite.Secrets.Owner != nil {
```
Comment (tom-hadlaw-hs) [inline secrets/json_secrets.go:55]: Strings can't be nil, you can probably just check if it's empty here.

File: secrets/json_secrets.go [hunk 3]
```diff
@@ -52,6 +52,13 @@ func WriteComposite(composite config.CompositeSecretFile, cache briefcase.Secret
...
+        err = os.Chown(composite.Filename, composite.Secrets.Owner , composite.Secrets.Owner)
```
Comment (tom-hadlaw-hs) [inline secrets/json_secrets.go:56]: I think os.Chown takes a integer GID + UID, you'll probably need to do a lookup for those.

**Comment by shuai-hao-hs** [comment]
This is a rare case (only for puppet) and it’s taken care of in the ansible. Close this PR as the work is not scheduled in the near future.

**Pull request closed without merging**